### PR TITLE
451001 oksana piskunova

### PR DESCRIPTION
**Pull request by OksanaPiskunova**

### CODE REVIEW

File: WebCrawler/WebCrawler/HtmlParser/SimpleHtmlParser.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+using System.Threading.Tasks;
+
+namespace WebCrawler.HtmlParser
+{
+    public class HtmlParser : IHtmlParser
+    {
+        private const string LinkTag = "a";
+        private const string HrefAttribute = "href";
+
+        public ConcurrentBag<Exception> ParserErrors { get; private set; }
```
Comment (RamanBut-Husaim) [inline WebCrawler/WebCrawler/HtmlParser/SimpleHtmlParser.cs]: @OksanaPiskunova Why are you using a `Concurrent` collection there?

File: WebCrawler/WebCrawler/WebCrawler/SimpleWebCrawler.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+using NLog;
+using WebCrawler.HtmlParser;
+using WebCrawler.WebCrawlerResults;
+
+namespace WebCrawler.WebCrawler
+{
+    public class SimpleWebCrawler : ISimpleWebCrawler
+    {
+        private readonly IHtmlParser _htmlParser;
+        private readonly Logger _logger;
```
Comment (RamanBut-Husaim) [inline WebCrawler/WebCrawler/WebCrawler/SimpleWebCrawler.cs]: @OksanaPiskunova As you create a logger for a class please consider making this field static as well.

File: WebCrawler/WebCrawlerTest/Model/WebCrawlerModel.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace WebCrawlerTest.Model
+{
+    internal class WebCrawlerModel
+    {
+        private const string ConfigFileName = "config.xml";
+
+        private readonly IConfigReader _configReader;
+        private readonly ISimpleWebCrawler _webCrawler;
+        private readonly Logger _logger;
```
Comment (RamanBut-Husaim) [inline WebCrawler/WebCrawlerTest/Model/WebCrawlerModel.cs]: @OksanaPiskunova The same issue with logger field.

File: WebCrawler/WebCrawlerTest/Properties/Annotations.cs [hunk 4]
```diff
@@ -0,0 +1,1039 @@
+﻿/* MIT License
```
Comment (RamanBut-Husaim) [inline WebCrawler/WebCrawlerTest/Properties/Annotations.cs]: @OksanaPiskunova For what purpose are you using this file?

File: WebCrawler/WebCrawlerTest/ViewModel/WebCrawlerViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                    if (ClickingCommand.CanExecute)
+                    {
+                        ClickingCommand.CanExecute = false;
+                        ClickCount++;
+                        ClickingCommand.CanExecute = true;
+                    }
+                });
+        }
+
+        protected virtual void OnPropertyChanged([CallerMemberName] string propertyName = null)
```
Comment (RamanBut-Husaim) [inline WebCrawler/WebCrawlerTest/ViewModel/WebCrawlerViewModel.cs]: @OksanaPiskunova C#6 includes an amazing `nameof` operator. Do you really think that now you should use this attribute?

File: WebCrawler/WebCrawlerTest/Properties/Annotations.cs (hunk 4 above)
Comment (RamanBut-Husaim) [inline WebCrawler/WebCrawlerTest/Properties/Annotations.cs]: cpy_pst

File: WebCrawler/WebCrawler/HtmlParser/SimpleHtmlParser.cs (hunk 1 above)
Comment (OksanaPiskunova) [inline WebCrawler/WebCrawler/HtmlParser/SimpleHtmlParser.cs]: > Concurrent collections in .NET work very much like their single-thread counterparts with the difference that they are thread safe. These collections can be used in scenarios where you need to share a collection between Tasks. They are typed and use a lightweight synchronization mechanism to ensure that they are safe and fast to use in parallel programming.

**Comment by OksanaPiskunova** [comment]
Please, take a look.

File: WebCrawler/WebCrawler/HtmlParser/SimpleHtmlParser.cs (hunk 1 above)
Comment (RamanBut-Husaim) [inline WebCrawler/WebCrawler/HtmlParser/SimpleHtmlParser.cs]: @OksanaPiskunova Agree. :) And where do you parallelize your code?

**Pull request merged**